#11813: Fixing V No. 1

oarrietadotcms opened the pull request:
**ContentTypeFactoryImp.java**: Sanitize condition and search before adding to DotConnect.
**SQLUtil.java**: Separate Condition and Parameter evil words just as the sanitize methods.
**SQLUtilTest.java**: Add tests to check new condition and parameter evil words and methods.

oarrietadotcms committed:
#11813: Fixing V No. 1
ContentTypeFactoryImp.java: Sanitize condition and search before adding to DotConnect.
SQLUtil.java: Separate Condition and Parameter evil words just as the sanitize methods.
SQLUtilTest.java: Add tests to check new condition and parameter evil words and methods.

wezell commented on dotCMS/src/main/java/com/dotcms/contenttype/business/ContentTypeFactoryImpl.java:
```diff
@@ -526,12 +507,12 @@ private boolean dbDelete(ContentType type) throws DotDataException {
-    dc.addParam(searchCondition.search);
-    dc.addParam(searchCondition.search);
-    dc.addParam(searchCondition.search);
+    dc.addParam( SQLUtil.sanitizeParameter( searchCondition.search ) );
```
We should not need to sanitize a parameter in a prepared statement, the drive should take care of this for us, right?

dsilvam replied:
this is correct, not needed, only the sort by and searchCondition.condition replaced in the sql

jdotcms replied:
Good catch, if the parameter is with ?, the driver will escape invalid characters

oarrietadotcms replied:
you are right, as long as we use addParam method we don't need to sanitize it. Let me update the source code.

oarrietadotcms committed:
#11813: Feedback from Code Review:
ContentTypeFactoryImp.java: As long as we use addParam method we don't need to sanitize it. Driver will take care of that.

dsilvam commented on dotCMS/src/test/java/com/dotmarketing/common/util/SQLUtilTest.java:
```diff
@@ -550,5 +551,44 @@ public void testValidOr2SanitizeParameter() throws Exception {
+    public void testValidCondition() throws Exception {
+
+        final String query = "structuretype = 1";
+        final String s = SQLUtil.sanitizeParameter( query );
```
should this call SQLUtil. sanitizeCondition instead?

dsilvam commented on dotCMS/src/test/java/com/dotmarketing/common/util/SQLUtilTest.java:
```diff
@@ -550,5 +551,44 @@ public void testValidOr2SanitizeParameter() throws Exception {
+    public void testInvalidCondition() throws Exception {
+
+        final String query = "and if(length(user())>0,sleep(10),2)";
+        final String s = SQLUtil.sanitizeParameter( query );
```
SQL.sanitizeCondition also?

oarrietadotcms committed:
#11813: Feedback from Code Review:
SQLUtilTest.java: Test should be using sanitizeCondition method.

oarrietadotcms commented:
PR updated with code review feedback.
